File: app/src/main/java/com/savvasdalkitsis/gameframe/main/presenter/MainPresenter.java
package com.savvasdalkitsis.gameframe.main.presenter;

import com.savvasdalkitsis.gameframe.ip.repository.IpRepository;
import com.savvasdalkitsis.gameframe.main.view.MainView;
import com.savvasdalkitsis.gameframe.rx.RxTransformers;

public class MainPresenter {

    private final IpRepository ipRepository;
    private MainView mainView;

    public MainPresenter(IpRepository ipRepository) {
        this.ipRepository = ipRepository;
    }

    public void bindView(MainView mainView) {
        this.mainView = mainView;
    }

    public void loadIpAddress() {
        ipRepository.getIpAddress()
                .compose(RxTransformers.schedulers())
                .subscribe(mainView::ipAddressLoaded, mainView::ipCouldNotBeFound);
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/main/presenter/MainPresenter.kt
package com.savvasdalkitsis.gameframe.main.presenter

import com.savvasdalkitsis.gameframe.ip.model.IpAddress
import com.savvasdalkitsis.gameframe.ip.repository.IpRepository
import com.savvasdalkitsis.gameframe.main.view.MainView
import com.savvasdalkitsis.gameframe.rx.RxTransformers

class MainPresenter(private val ipRepository: IpRepository) {
    private var mainView: MainView? = null

    fun bindView(mainView: MainView) {
        this.mainView = mainView
    }

    fun loadIpAddress() {
        ipRepository.ipAddress
                .compose<IpAddress>(RxTransformers.schedulers<IpAddress>())
                .subscribe(
                        { mainView?.ipAddressLoaded(it) },
                        { mainView?.ipCouldNotBeFound(it) }
                )
    }
}
